flink-ml-lib/src/main/java/org/apache/flink/ml/common/param/HasK.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.ml.common.param;
+
+import org.apache.flink.ml.param.IntParam;
+import org.apache.flink.ml.param.Param;
+import org.apache.flink.ml.param.ParamValidators;
+import org.apache.flink.ml.param.WithParams;
+
+/** Interface for the shared K param. */
+public interface HasK<T> extends WithParams<T> {
+
+    /**
+     * topK
+     */
+    Param<Integer> K = new IntParam("k", "k", 10, ParamValidators.gt(0));
```

Comment:
If an agreement has been reached that we should extract `K` as a params, I suggest that we can directly apply this to all algorithms that have implemented their own `getK` `setK` methods.

Besides, it might be better to set the description, default value and validators in reference to existing practices.

Reply:
OK, I will refine other use case.

Reply:
done